newrelic-integration-e2e/pkg/dockercompose/docker.go

```diff
@@ -4,59 +4,90 @@ import (
 	"fmt"
 	"os"
 	"os/exec"
+
+	"github.com/newrelic/newrelic-integration-e2e-action/newrelic-integration-e2e/internal/runtime/logger"
 )
 
 const (
 	dockerComposeBin = "docker-compose"
 	dockerBin        = "docker"
 )
 
-func Run(path string, container string, envVars map[string]string) error {
-	if err := Build(path, container, envVars); err != nil {
-		return err
-	}
-	args := []string{"-f", path, "run"}
-	for k, v := range envVars {
-		args = append(args, "-e", fmt.Sprintf("%s=%s", k, v))
+type Compose struct {
+	path      string
+	env       map[string]string
+	buildArgs map[string]string
+
+	cmdLogger logger.CommandLogger
+}
+
+// New creates a new docker-compose wrapper given the path to a docker-compose.yml file.
+func New(filepath string) *Compose {
+	return &Compose{
+		path:      filepath,
+		cmdLogger: logger.NewGHALogger(os.Stderr),
```

Comment:
Ideally yes, but unfortunately the calling chain  from Runner -> Agent -> Compose made this very hard. Notably the fact that the `agent` receives a config interface (??) rather than a struct, so was a bit painful to add all the required config there.

Since it previously was simply logged to stderr, I don't think this is a downgrade. I'd vote for adding the config flag in a follow up pr if we think it's necessary.